api/src/main/java/io/kafbat/ui/config/auth/BasicAuthSecurityConfig.java

```diff
@@ -42,4 +60,42 @@ public SecurityWebFilterChain configure(ServerHttpSecurity http) {
     return builder.build();
   }
 
+  @Bean
+  public ReactiveUserDetailsService reactiveUserDetailsService(SecurityProperties properties,
+                                                               ObjectProvider<PasswordEncoder> passwordEncoder,
+                                                               AccessControlService accessControlService) {
+    SecurityProperties.User user = properties.getUser();
+
+    UserDetails userDetails = User.withUsername(user.getName())
+        .password(password(user.getPassword(), passwordEncoder.getIfAvailable()))
+        .roles(StringUtils.toStringArray(user.getRoles()))
+        .build();
+
+    Collection<String> groups = accessControlService.getRoles().stream()
+        .filter(role -> role.getSubjects().stream()
+            .filter(subj -> Provider.BASIC_AUTH.equals(subj.getProvider()))
```

Comment:
this looks like an easy implementation, but I don't quite like the idea of having RBAC related business logic in spring security config-related class. We do extract RBAC roles in extractors, I think we should do the same here as well.